src/dmd/builtin.d

```diff
@@ -25,8 +25,43 @@ import dmd.mtype;
 import dmd.root.ctfloat;
 import dmd.root.stringtable;
 import dmd.tokens;
+import dmd.id;
 static import core.bitop;
 
+/**********************************
+ * Determine if function is a builtin one that we can
+ * evaluate at compile time.
+ */
+public extern (C++) BUILTIN isBuiltin(FuncDeclaration fd)
+{
+    if (fd.builtin == BUILTIN.unknown)
+    {
+        fd.builtin = determine_builtin(fd);
+    }
+    return fd.builtin;
+}
+
+/**************************************
+ * Evaluate builtin function.
+ * Return result; NULL if cannot evaluate it.
+ */
+public extern (C++) Expression eval_builtin(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    if (fd.builtin == BUILTIN.unknown)
+        fd.builtin = determine_builtin(fd);
+    if (fd.builtin == BUILTIN.unimp)
```

Comment:
This test should be first as it is most popular.

Reply:
Can't do that. We need to test if it has been tested before. Note that `determine_builtin` may well return `BUILTIN.unimp`